ansible/TestbedProcessing.py

```diff
@@ -484,47 +477,59 @@ def makeLab(data, devices, testbed, outfile):
-                        except AttributeError:
-                            print("\t\t" + host + " switch_type not found")
+                                entry += "\tswitch_type=" + str( switch_type )
+                                if switch_type == 'voq' and card_type != 'supervisor':
+                                    if num_asics is None:
+                                        num_asics = 1
+
+                                    # All fields are a list. For single asic the list is of size 1.
+                                    switchids = dev.get("switchids")                # switchids
+                                    voq_inband_ip = dev.get("voq_inband_ip")        # voq_inband_ip
+                                    voq_inband_ipv6 = dev.get("voq_inband_ipv6")    # voq_inband_ipv6
+                                    voq_inband_intf = dev.get("voq_inband_intf")    # voq_inband_intf
+                                    voq_inband_type = dev.get("voq_inband_type")    # voq_inband_type
+                                    max_cores = dev.get("max_cores")                # max cores
+                                    if num_asics > 1:
+                                        lo4096_ip = dev.get("loopback4096_ip")      # loopback4096_ip
+                                        lo4096_ipv6 = dev.get("loopback4096_ipv6")  # loopback4096_ipv6
+
+                                    # Set defaults
+                                    if switchids is None:
+                                        switchids = [start_switchid + asic_id for asic_id in range(num_asics)]
+                                    if voq_inband_ip is None:
+                                        voq_inband_ip = ["1.1.1.{}/32".format(start_switchid + asic_id) for asic_id in range(num_asics)]
+                                    if voq_inband_ipv6 is None:
+                                        voq_inband_ip = ["1111::1:{}/128".format(start_switchid + asic_id) for asic_id in range(num_asics)]
+                                    if voq_inband_intf is None:
+                                        voq_inband_intf = ["Ethernet-IB{}".format(asic_id) for asic_id in range(num_asics)]
+                                    if voq_inband_type is None:
+                                        voq_inband_type = "port"
+                                    if max_cores is None:
+                                        max_cores = 48
+                                    if num_asics > 1:
+                                        if lo4096_ip is None:
+                                            lo4096_ip = ["8.0.0.{}/32".format(start_switchid + asic_id) for asic_id in range(num_asics)]
+                                        if lo4096_ipv6 is None:
+                                            lo4096_ipv6 = ["2603:10e2:400::{}/128".format(start_switchid + asic_id) for asic_id in range(num_asics)]
+
+                                    start_switchid += num_asics
```

Comment:
number of cores per asic have to be taken into account when calculating the switchid.
so it would be like
                              start_switchid += num_asics * num_cores_per_asic
num_cores_per_asic could be assumed to be same for all linecards for now if that is the case.
Else it also has to be a list

Reply:
resolved in commit https://github.com/Azure/sonic-mgmt/pull/3746/commits/c8602e1275e93ac7c2febbabfa543c9725e60bc7